### PR TITLE
SetupOutputVariable wrapper to use enum type parameters (endUseCategory, ReportingFrequency, and eResouceType)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - An initial step following up the comment received in Issue #10228 about `enum` parematers for endUseCategory (and also ReportingFrequency, and eResourceType), which could potentially enforce more rigorous check by default, and reduce the sanity checks down the road;
 - The approach here is to carefully engineer a few overload function wrapper as an interpreter (or driver) to call `SetupOutputVariable()` either with more `enum` alternative in the parameter list, or via the original way that maintains compataiabilty with the existing code. The old way of calling it is unaffected, which will allow a smoother transition to the new method (if it is the way to go).
 - The final call would be directed to the current `SetupOutputVariable()` code, which has been subjected under longer scrutiny and fixes along the way. The overloaded driver layer provides a separate place for conducting a good initials sanity checks for input parameters;
 - Currently, this goal is obtained: `SetupOutputVariable` can be used either the new (`enum`) way or the original way. It is encouraged to use the new way though, since it would be allow the input values to have better sanity checks.
 -  In the process, though, the key is to carefully divide the current functions into different categories; and also to deal with some corner cases. You may also notice in the code that a few `enum` definitions were definite in a slightly different way (but values are pretty much the same) to keep computability with the existing code calls---it's actually after some careful consideration to go this way.
 - The transition to use the new `enum` call were started (and demonstrated) in about 4-5 files.
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/PluginManager.cc [hunk 1]
```diff
@@ -201,7 +201,7 @@ void PluginManager::setupOutputVariables([[maybe_unused]] EnergyPlusData &state)
                                         sUpdateFreq,
                                         sAvgOrSum,
                                         thisObjectName,
-                                        {},
+                                        "",
```
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.cc]: Is this a better option?  All the changes from strings to enums look great, this just seems like an out of place change.  Could you clarify?

Comment (jcyuan2020) [reply src/EnergyPlus/PluginManager.cc]: @Myoldmopar  This one can now be reverted (and also for another location for similar scenarios). I changed this during the middle of the development when at that time I indeed needed a parameter difference to distinguish the call. But now it is no longer needed.

**Comment by Myoldmopar** [comment]
Thanks for fixing that up @jcyuan2020, I'll run regressions locally and see if everything is good.  If so this can be ready to merge and we can hopefully eliminate the string interface soon.

**Comment by Myoldmopar** [comment]
I was half expecting this to be conflicted with that big PR merge, but nope!  I'll build and test, and if happy this will go in next.

**Comment by Myoldmopar** [comment]
It wasn't conflicted, but it did need a fix to the unit test for the new namespace.   CI actually just reported a couple custom check failures.  I'll try to fix them up.

**Comment by Myoldmopar** [comment]
CI is completely green here, thanks for tidying up those last pieces.  @amirroth has much, much deeper a refactor in output processor, but per his guidance, this can go in as is.  We won't waste any more effort or time here then, merging this.  Thanks @jcyuan2020

**Pull request merged**